src/subscription_manager/cache.py

```diff
@@ -237,7 +231,7 @@ def load_status(self, uep, uuid):
             raise
         # all of the abover are subclasses of ConnectionException that
         # get handled first
-        except (connection.ConnectionException,
+        except (connection.ConnectionException, connection.RateLimitExceededException,
```

Comment:
I'd really like to see a unit test for this.  Could you mock out whatever `_sync_with_server` is doing to raise a RateLimitExceededException and then check to make sure calls are made to `_cache_exists`/`_read_cache`?

Reply:
I just added one unit test covering this case.